File: Leetcode/1695.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        unordered_set<int> myset;
        int len = nums.size();
        int ret = 0, sum = 0;
        int left = 0, right = 0;
        
        while(right < len)
        {
            if(myset.count(nums[right]))
            {
                sum -= nums[left];
                myset.erase(nums[left]);
                ++left;
            }
            else
            {
                sum += nums[right];
                myset.insert(nums[right]);
                ++right;
            }
            
            ret = max(ret, sum);
        }
        
        return ret;
    }
};

File: Leetcode/1695.go
func maximumUniqueSubarray(nums []int) int {
    var left, right int = 0, 0
    var ans, score int = 0, 0
    set := make(map[int]bool)
    for ; right < len(nums) ; right++ {
        for set[nums[right]] {
            set[nums[left]] = false
            score -= nums[left]
            left++
        }
        set[nums[right]] = true
        score += nums[right]
        if score > ans {
            ans = score
        }
    }
    return ans
}